.github/scripts/sync_distributed_folder_prototype.sh

```diff
@@ -20,7 +20,12 @@ do
         continue
     fi
     echo "Copying $SHA"
-    git cherry-pick -x "$SHA"
+    git cherry-pick -x "$SHA" -X theirs
+    git reset --soft HEAD~1
+    git add torch/distributed torch/csrc/distributed test/distributed test/cpp/c10d benchmarks/distributed
```

Comment:
Are we worried that only doing this might set us up for longer term failures?

i.e. are we absolutely sure that commits that go to these folders are isolated and aren't dependent on changes elsewhere?

Reply:
I'm a bit worried, but I think we'll just need to try it out and see what happens.

Reply:
I looked through the commit log and the vast majority of changes are due to lint or formatting changes rather than "real" changes, so I'm cautiously optimistic that this will be ok